src/js/components/form-elements/FieldArrayComponent.jsx

```diff
@@ -45,21 +46,34 @@ class FieldArrayComponent extends Component {
             </tr>
           </thead>
           <tbody>
-            {fields.map((field, index) => (
-            // eslint-disable-next-line react/no-array-index-key
-              <tr key={index}>
-                { _.map(fieldsConfig.fields, (config, name) => (
-                  <td key={`${field}.${name}`} className="align-middle">
-                    { renderFormField(config, `${field}.${name}`, {
-                    ...properties,
-                    arrayField: true,
-                    addRow,
-                    removeRow: () => fields.remove(index),
-                    rowIndex: index,
-                  })}
-                  </td>
-              )) }
-              </tr>))}
+            {fields.map((field, index) => {
+              if (fieldsConfig.rowWithSelector) {
+                return (
+                  <TableRowWithSelector
```

Comment:
this can be parametrized like with AddButton, so we can use more types of Row components, also the code duplication will be removed